frontend/components/wizard/results-step/components/document-reconstructor.tsx

```diff
@@ -11,53 +13,254 @@ interface DocumentReconstructorProps {
+  selectedChunkIndex: number | null;
+  onChunkSelect: (chunkIndex: number | null) => void;
+}) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const combinedContent = useMemo(() => {
+    if (chunks.length === 0) {
+      return null;
+    }
+
+    // Detect block-level syntax from the first chunk
+    const firstChunk = chunks[0];
+    const blockPrefix = detectBlockSyntax(firstChunk.content);
+
+    // Extract content from all chunks and wrap in spans
+    const wrappedChunks = chunks
+      .map((chunk) => {
+        const content = extractChunkContent(chunk.content, blockPrefix);
+        const severity = getSeverity(results, chunk);
+        return `<span data-chunk-index="${chunk.chunkIndex}" data-severity="${severity}">${content}</span>`;
+      })
+      .join(' ');
+
+    // Reconstruct markdown with block-level syntax from first chunk
+    return blockPrefix ? `${blockPrefix}${wrappedChunks}` : wrappedChunks;
+  }, [chunks, results]);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return;
+    }
+    const targets = containerRef.current.querySelectorAll('[data-chunk-index]');
+    targets.forEach((target: Element) => {
+      if (selectedChunkIndex !== null) {
+        target.setAttribute(
+          'data-chunk-selected',
+          selectedChunkIndex === parseInt(target.getAttribute('data-chunk-index') || '0') ? 'true' : 'false',
+        );
+      } else {
+        target.removeAttribute('data-chunk-selected');
+      }
+    });
+  }, [selectedChunkIndex]);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return;
+    }
+
+    const handleClick = (event: Event) => {
+      const target = event.target as HTMLElement;
+      const chunkIndex = target.getAttribute('data-chunk-index');
+      if (chunkIndex && onChunkSelect) {
+        onChunkSelect(parseInt(chunkIndex));
+      }
+    };
+
+    const targets = containerRef.current.querySelectorAll('[data-chunk-index]');
+    targets.forEach((target: Element) => {
+      target.addEventListener('click', handleClick);
+    });
+    return () => {
+      targets.forEach((target: Element) => {
+        target.removeEventListener('click', handleClick);
+      });
+    };
+  }, [combinedContent, onChunkSelect]);
+
+  return (
+    <div
+      ref={containerRef}
+      className={cn(
+        "[&_[data-severity='high']]:bg-red-100",
+        "[&_[data-severity='medium']]:bg-yellow-100",
+        "[&_[data-severity='low']]:bg-blue-100",
+        '[&_[data-severity]]:hover:bg-gray-300/50 [&_[data-severity]]:cursor-pointer',
+        '[&_[data-chunk-selected="true"]]:shadow-lg',
+        '[&_[data-chunk-selected="false"]]:opacity-50',
+      )}
+    >
+      <Markdown rehypePlugins={[rehypeRaw]}>{combinedContent}</Markdown>
     </div>
   );
 }
+
+/**
+ * Detects block-level markdown syntax from a chunk.
+ * Returns the block prefix (e.g., "# ", "- ", "> ") or null for paragraphs.
+ */
+function detectBlockSyntax(chunk: string): string | null {
```

Comment:
I extracted this to utils files, it might be good to rebase your branch with main and check the conflicts